diff --git a/core/lib/Drupal/Component/EventDispatcher/ContainerAwareEventDispatcher.php b/core/lib/Drupal/Component/EventDispatcher/ContainerAwareEventDispatcher.php
index 323b8e37f2..dae9c8b356 100644
--- a/core/lib/Drupal/Component/EventDispatcher/ContainerAwareEventDispatcher.php
+++ b/core/lib/Drupal/Component/EventDispatcher/ContainerAwareEventDispatcher.php
@@ -140,7 +140,9 @@ public function dispatch($event/*, string $event_name = NULL*/) {
           }
 
           call_user_func($definition['callable'], $event, $event_name, $this);
-          if ($event->isPropagationStopped()) {
+          $stoppable = method_exists($event, 'isPropagationStopped');
+
+          if ($stoppable && $event->isPropagationStopped()) {
             return $event;
           }
         }
